src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,601 @@
+    quad_points_a = getpoints(iv.here.qr, face_here)
+    quad_points_b = getpoints(iv.there.qr, face_there)
+    transform_interface_points!(quad_points_b, quad_points_a, interface_transformation)
+    @boundscheck checkface(iv.there, face_there)
+    # TODO: This is the bottleneck, cache it?
+    @assert length(quad_points_a) <= length(quad_points_b)
+    # Re-evalutate shape functions in the transformed quadrature points
+    for qp in 1:length(quad_points_a), i in 1:getnbasefunctions(iv.there)
+        iv.there.dNdξ[i, qp, face_there], iv.there.N[i, qp, face_there] = shape_gradient_and_value(iv.there.func_interp, quad_points_b[qp], i)
+    end
+    for qp in 1:length(quad_points_a), i in 1:getngeobasefunctions(iv.there)
+        iv.there.dMdξ[i, qp, face_there], iv.there.M[i, qp, face_there] = shape_gradient_and_value(iv.there.geo_interp, quad_points_b[qp], i)
+    end
+    # reinit! the "there" side
+    reinit!(iv.there, coords_there, face_there)
+    return iv
+end
+
+"""
+    getnormal(iv::InterfaceValues, qp::Int; here::Bool=true)
+
+Return the normal vector in the quadrature point `qp` on the interface. If `here = true`
+(default) the outward normal to the "here" element is returned, otherwise the outward normal
+to the "there" element.
+"""
+function getnormal(iv::InterfaceValues, qp::Int; here::Bool=true)
+    # TODO: Does it make sense to allow the kwarg here? You can juse negate the vector
+    #       yourself since getnormal(iv, qp; here=false) == -getnormal(iv, qp; here=true).
+    return here ? iv.here.normals[qp] : iv.there.normals[qp]
+end
+
+"""
+    function_value(iv::InterfaceValues, q_point::Int, u_here::AbstractVector, u_there::AbstractVector; here::Bool)
+    function_value(iv::InterfaceValues, q_point::Int, u_here::AbstractVector, range_here, u_there::AbstractVector, range_there; here::Bool)
+
+Compute the value of the function in quadrature point `q_point` on the "here" (`here=true`)
+or "there" (`here=false`) side of the interface. `u_here` and `u_there` are the values of
+the degrees of freedom for the respeciv element.
+
+
+TODO: Complete the docs here.
+is a vector with values
+for the degrees of freedom. For a scalar valued function, `u` contains scalars.
+For a vector valued function, `u` can be a vector of scalars (for use of `VectorValues`)
+or `u` can be a vector of `Vec`s (for use with ScalarValues).
+
+`here` determines which element to use for calculating function value.
+`true` uses the element A's nodal values, which is the default, while `false` uses element B's.
+
+The value of a scalar valued function is computed as ``u(\\mathbf{x}) = \\sum\\limits_{i = 1}^n N_i (\\mathbf{x}) u_i``
+where ``u_i`` are the value of ``u`` in the nodes. For a vector valued function the value is calculated as
+``\\mathbf{u}(\\mathbf{x}) = \\sum\\limits_{i = 1}^n N_i (\\mathbf{x}) \\mathbf{u}_i`` where ``\\mathbf{u}_i`` are the
+nodal values of ``\\mathbf{u}``.
+"""
+function_value(::InterfaceValues, ::Int, args...)
```

Comment:
@KnutAM 
> > Good point. I guess we can do
> > ```
> > function_(value|gradient|...)(iv::InterfaceValues, qp::Int, ue::Vector{Float64}, dof_range::UnitRange, dof_range::UnitRange)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > then and let the user handle the offset. It is easy to catch the mistake of not ofsetting in many cases I think, so should be reasonably safe.
> 
> Has there been any new discussion on this topic, also following [#743 (comment)](https://github.com/Ferrite-FEM/Ferrite.jl/pull/743#issuecomment-1629260683) (at least to me, it looks like the current implementation is different)?
> 
> For being able to use automatic differentiation, I think a single dof-vector, `ue`, input would be most practical.

IIRC @fredrikekre  noted that we might sometimes have to provide two different dof ranges, so going with two `u` s would make more sense, right?
I think the other options are
- a single `u` and no dof ranges -> first half of `u` is here and the other half is `there` (This can be broken for mixed elements)
- a single `u` and two dof ranges

(Leaving this as a code comment so we can reply to it)

Reply:
For single-field problems (or when giving only the dofs corresponding to a single field), the interpolation information should be enough to know how many dofs belong to here and there, even if they are different, right?

For multi-field problems (analogous to the current dof range approach), then two dof-ranges are required. And the main challenge here, is that the user needs to offset the "there" dof-range? (Unless the number of celldofs for "here" is provided in the `reinit!`, which should be feasible since it should be available in the cache).

Reply:
The AD argument is good. I suppose there should be some method of `dof_range` that returns two, and the second one with an offset.